runner/runners/single_test.go

```diff
@@ -105,6 +105,34 @@ func TestAbort(t *testing.T) {
 	}
 }
 
+func TestAbortLogUpload(t *testing.T) {
+	defer teardown(t)
+	sim := execers.NewSimExecer()
+	outputCreator, err := NewHttpOutputCreator("")
+	if err != nil {
+		panic(err)
+	}
+	filerMap := runner.MakeRunTypeMap()
+	filerMap[runner.RunTypeScoot] = snapshot.FilerAndInitDoneCh{Filer: snapshots.MakeInvalidFiler(), IDC: nil}
+	r := NewSingleRunner(sim, filerMap, outputCreator, nil, stats.NopDirsMonitor, runner.EmptyID, []func() error{}, []func() error{}, NewWaitingLogUploader())
+	args := []string{"complete 0"}
+	runID := run(t, r, args)
+	assertWait(t, r, runID, running(), args...)
+	time.Sleep(10 * time.Millisecond)
```

Comment:
Is this prone to fail or skip the upload without the Sleep here? Wonder if asa an alternative we could add an additional channel to NoopWaitingLogUploader such that UploadLog would wait to receive on before sending or closing its cancelCh, and we could signal that additional channel here, verifying we actually hit that code as intended. WDYT?

Reply:
Yeah, that's a good idea, will update this test. Had added sleep to make sure we're not aborting too early and skipping log upload.